pkg/prometheus/rules.go

```diff
@@ -332,20 +332,37 @@ func prometheusRuleConfigMapName(prometheusName string) string {
 }
 
 // GenerateContent takes a PrometheusRuleSpec and generates the rule content
-func GenerateContent(promRule monitoringv1.PrometheusRuleSpec, logger log.Logger) (string, error) {
+func GenerateContent(objectType string, promRule monitoringv1.PrometheusRuleSpec, logger log.Logger) (string, error) {
 	content, err := yaml.Marshal(promRule)
 	if err != nil {
 		return "", errors.Wrap(err, "failed to marshal content")
 	}
+
 	errs := ValidateRule(promRule)
-	if len(errs) != 0 {
+	if errs != nil {
```

Comment:
Better to keep the original case IMHO as an empty slice might not be nil.

```suggestion
	if len(errs) != 0 {
```